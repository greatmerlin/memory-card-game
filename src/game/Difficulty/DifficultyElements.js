import styled from "styled-components";
import { Link as LinkRouter } from "react-router-dom";

export const Container = styled.div`
  min-height: 200px;
  padding: 5em;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  z-index: 0;
  overflow: hidden;
  background: linear-gradient(
    108deg,
    rgba(1, 147, 86, 1) 0%,
    rgba(10, 201, 122, 1) 100%
  );
`;

export const ContainerWin = styled.div`
  min-height: 200px;
  padding: 5em;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  z-index: 0;
  overflow: hidden;
  background: #282c34;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const ContainerWinText = styled.div`
  color: #fff;
`;

export const WinButton = styled(LinkRouter)`
  border: #282c34;
  border-style: solid;
  border-width: 5px;
  border-radius: 50px;
  background: #01bf71;
  color: white;
  padding: 10px;
  text-align: center;
  font-size: 16px;
  text-decoration: none;
  white-space: nowrap;
  outline: none;

`;

export const FormWrap = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  @media screen and (max-width: 400px) {
    height: 80%;
  }
`;

export const Icon = styled(LinkRouter)`
  margin-left: 32px;
  margin-top: 32px;
  text-decoration: none;
  color: #fff;
  font-weight: 700;
  font-size: 32px;
  padding: 1em;

  @media screen and (max-width: 480px) {
    margin-left: 16px;
    margin-top: 8px;
  }
`;

export const FormContent = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  @media screen and (max-width: 480px) {
    padding: 10px;
  }
`;

export const Form = styled.form`
  background: #010101;
  max-width: 400px;
  height: auto;
  width: 100%;
  display: grid;
  margin: 0 auto;
  z-index: 1;
  padding: 80px 32px;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.9);

  @media screen and (max-width: 400px) {
    padding: 32px 32px;
  }
`;

export const FormH1 = styled.h1`
  margin-bottom: 40px;
  color: #fff;
  font-size: 20px;
  font-weight: 400;
  text-align: center;
`;

export const FormBackButton = styled(LinkRouter)`
  background: #01bf71;
  padding: 16px 0;
  margin-top: 20px;
  border-radius: 4px;
  color: #fff;
  font-size: 20px;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
`;
