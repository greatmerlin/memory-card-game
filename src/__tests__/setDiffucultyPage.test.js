import React, { useState } from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import SetDifficultyPage from '../pages/setDifficultyPage.js';
import { FormH1 } from '../game/Difficulty/DifficultyElements.js';

configure({ adapter: new Adapter() });

// Testing area



it("renders setDifficulty page without crashing", () => {
    shallow(<SetDifficultyPage />);
  });
  
  it("setDifficulty components renders correctly", () => {
    const tree = shallow(<SetDifficultyPage />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("renders setDifficulty header and called the state correctly", () => {
    const [playerName, setplayerName] = useState("");
    setplayerName("Theo");
    const wrapper = shallow(<SetDifficultyPage />);
    const welcome = <FormH1>Hello Theo, select the difficulty</FormH1>;
    expect(wrapper.contains(welcome)).toEqual(true);
  });
  