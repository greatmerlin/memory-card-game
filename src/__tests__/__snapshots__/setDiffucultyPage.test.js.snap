// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setDifficulty components renders correctly 1`] = `
<Fragment>
  <Sidebar
    isOpen={false}
    toggle={[Function]}
  />
  <SilentNavbar
    toggle={[Function]}
  />
  <styled.div>
    <styled.div>
      <Styled(Link)>
        Memory UP!
      </Styled(Link)>
      <styled.div>
        <styled.form>
          <styled.h1>
            Hello undefined, select the difficulty
          </styled.h1>
          <Styled(Link)
            to={
              Object {
                "pathname": "/game",
                "state": Object {
                  "difficulty": "easy",
                },
              }
            }
          >
            Easy
          </Styled(Link)>
          <Styled(Link)
            to={
              Object {
                "pathname": "/game",
                "state": Object {
                  "difficulty": "medium",
                },
              }
            }
          >
            Medium
          </Styled(Link)>
          <Styled(Link)
            to={
              Object {
                "pathname": "/game",
                "state": Object {
                  "difficulty": "hard",
                },
              }
            }
          >
            Hard
          </Styled(Link)>
        </styled.form>
      </styled.div>
    </styled.div>
  </styled.div>
  <Footer />
</Fragment>
`;
